src/V3EmitC.cpp

```diff
@@ -1223,6 +1223,45 @@ class EmitCStmts : public EmitCBaseVisitor {
         }
     }
 
+    void emitStdRandomizePartial(AstNode* argp, AstNodeDType* dtypep) {
+        if (auto enumDType = VN_CAST(dtypep, EnumDType)) {
```

Comment:
I suspect enum randomization will be the most controversial change here, as it puts lots of values on the stack and randomly selects one of them, which isn't super fast. I would love to have a static array of possible enum values emitted somewhere and just pass that here, but I couldn't find a good way to do it.

Reply:
You're right I'd object ;)

V3Width::enumVarp can already make a list of enum value attributes, it's used to implement enum.first/last/next/prev.  You can extend it to make another static packed table.  Then this becomes a random index into that table.